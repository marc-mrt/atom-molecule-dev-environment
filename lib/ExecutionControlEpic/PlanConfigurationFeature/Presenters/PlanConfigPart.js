"use babel";
// @flow

import * as React from "react";
import PlanConfigSelectInputField from "./PlanConfigSelectInputField";
import PlanConfigTextInputField from "./PlanConfigTextInputField";
import PlanConfigGroup from "./PlanConfigGroup";
import type {
  ArrayConfigSchema,
  ConditionalConfigSchema,
  ConfigSchemaPartBase,
  EnumConfigSchema,
  NumberConfigSchema,
  ObjectConfigSchema,
  StringConfigSchema,
  FileConfigSchema,
} from "../Types/types.js.flow";
import PlanConfigConditionalGroup from "./PlanConfigConditionalGroup";
import PlanConfigNumberInputField from "./PlanConfigNumberInputField";
import PlanConfigCheckboxInputField from "./PlanConfigCheckboxInputField";
import PlanConfigFileInputField from "../Containers/PlanConfigFileInputField";
import PlanConfigList from "./PlanConfigList";

export default function PlanConfigPart({
  type,
  value,
  onChange,
  label,
  items,
  expression,
  cases,
  placeholder,
  schemas,
  tester,
  name,
  ...props
}: { name: string } & ConfigSchemaPartBase & {
    onChange: (value: any) => void,
    // eslint-disable-next-line prettier/prettier
  } & ( // eslint-disable-next-line prettier/prettier
    | (EnumConfigSchema & { value: any })
    | (FileConfigSchema & { value: any })
    | (ObjectConfigSchema & { value: { [key: string]: any } })
    | (StringConfigSchema & { value: string })
    | (NumberConfigSchema & { value: number })
    | (ConditionalConfigSchema & {
        value: { expressionValue: mixed, caseValue: mixed },
      })
    | (ArrayConfigSchema & { value: Array<any> })
  )) {
  switch (type) {
    case "enum":
      return (
        <PlanConfigSelectInputField
          options={props.enum}
          default={props.default}
          value={value}
          onChange={onChange}
          name={name}
          label={label}
        />
      );
    case "file":
      return (
        <PlanConfigFileInputField
          tester={tester}
          default={props.default}
          value={value}
          label={label}
          onChange={onChange}
          name={name}
        />
      );
    case "object":
      return (
        <PlanConfigGroup
          label={label}
          schemas={schemas}
          value={value}
          onChange={onChange}
          name={name}
        />
      );
    case "string":
      return (
        <PlanConfigTextInputField
          default={props.default}
          value={value}
          onChange={onChange}
          label={label}
          placeholder={placeholder}
          name={name}
        />
      );
    case "number":
      return (
        <PlanConfigNumberInputField
          default={props.default}
          value={value}
          onChange={onChange}
          label={label}
          name={name}
        />
      );
    case "conditional":
      return (
        <PlanConfigConditionalGroup
          default={props.default}
          expression={expression}
          cases={cases}
          value={value}
          onChange={onChange}
          name={name}
          label={label}
        />
      );
    case "array":
      return (
        <PlanConfigList
          value={value}
          default={props.default}
          onChange={onChange}
          items={items}
          label={label}
          name={name}
        />
      );
    case "boolean":
      return (
        <PlanConfigCheckboxInputField
          value={value}
          default={props.default}
          onChange={onChange}
          label={label}
          name={name}
        />
      );
    default:
      return null;
  }
}
