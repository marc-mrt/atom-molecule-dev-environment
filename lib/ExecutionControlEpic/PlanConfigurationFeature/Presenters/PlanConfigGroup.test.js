import "raf/polyfill";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import { PlanConfigGroup } from "./PlanConfigGroup";
import PlanConfigPart from "./PlanConfigPart";

Enzyme.configure({ adapter: new Adapter() });

describe("PlanConfigGroup", () => {
  it("should display a PlanConfigPart for every key of schemas", () => {
    let subject = shallow(
      <PlanConfigGroup
        schemas={{
          image: { type: "string" },
          scale: { type: "number" },
        }}
        value={{ image: "ubuntu", scale: 3 }}
      />,
    );

    expect(subject.find(PlanConfigPart).length).toBe(2);
    expect(subject).toMatchSnapshot();
  });
});
