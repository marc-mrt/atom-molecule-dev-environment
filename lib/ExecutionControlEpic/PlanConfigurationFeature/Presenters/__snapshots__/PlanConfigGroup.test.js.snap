// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanConfigGroup should display a PlanConfigPart for every key of schemas 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <styled.span>
      config
    </styled.span>
    <ReactImageFallback
      fallbackImage="arrow-right.svg"
      initialImage={null}
      src="atom://molecule-dev-environment/.storybook/public/arrow-right.svg"
      style={
        Object {
          "display": "flex",
          "height": "16px",
          "transform": "rotate(90deg)",
          "transition": "0.2s transform",
          "width": "16px",
        }
      }
    />
  </styled.div>
  <styled.div>
    <PlanConfigPart
      elementName="image"
      key="image"
      type="string"
    />
    <PlanConfigPart
      elementName="scale"
      key="scale"
      type="number"
    />
  </styled.div>
</styled.div>
`;
